library/Fission/User/CID/Query.hs

```diff
@@ -0,0 +1,24 @@
+module Fission.User.CID.Query
+  ( getUserCidsByUserId
+  , getUserCidsByCids
+  ) where
+
+import           Network.IPFS.CID.Types
+import           Database.Esqueleto hiding ((=.), update)
+
+import           Fission.Models
+import           Fission.Prelude
+
+-- | CIDs associated with a user
+getUserCidsByUserId :: MonadDB m => UserId -> Transaction m [Entity UserCid]
+getUserCidsByUserId userId =
+  select <| from \userCid -> do
+    where_ (userCid ^. UserCidUserFk ==. val userId)
+    return userCid
+
+-- | Find all CIDs that remain from a list
+getUserCidsByCids :: MonadDB m => [CID] -> Transaction m [Entity UserCid]
+getUserCidsByCids cids =
+  select <| from \userCid -> do
+    where_ (userCid ^. UserCidCid `in_` valList cids)
+    return userCid
```

Comment:
Same note one naming

Reply:
Also, while we're in here: we could think about turning these into classes to make mocking out in testing more granular.

```haskell
class MoandDB m => MonadDBQuery m model where
  getBy :: SqlExpr (Value Bool) -> Transaction [Entity model]

-- These can likely be defined in terms of `MonadQuery.get`

getAll
getAllById
-- and so on
```

Reply:
To hook into existing intuitions about why this would be helpful, we're essentially 12-factor/hexagonalizaing our app logic. Tests now can define their own version of `getBy` to return dummy data to do tests with zero DB setup/cleanup, really detailed tracing in development, and so on.

Correctness composes in this style, so if we test an endpoint in a dummy monad, and then test the production typeclass implementation, we know that the actual business logic works correctly. This is one of the advantages of pairing a strong type system with enforced purity.